### ISSUE TITLE
Use of list cols (simple features f.i.) slows the View 

### ISSUE THREAD

**Issue by RMHogervorst** [rstudio/rstudio#2039]
### System details

    OS Version      : both on windows amd ubuntu latest version of rstudio and R.
    RStudio Version :
    R Version       :

### Describe the problem in detail

When I open a dataframe in the viewer (View() ) it takes a long time to load. I would prefer if Rstudio noticed that this geography column doesn't need to be showed. 

I believe this has to do with loading the lists

What would be useful show  if a cell is empty, otherwise the word  geography or something in the cell .

**Comment by ateucher** [comment]
I posted a similar issue at [sf](https://github.com/r-spatial/sf/issues/618), with my workaround (basically hijack the `View()` function and strip off the sfc column). My question there was whether or not a fix would belong in **sf** and/or whether that should/could be behaviour **Rstudio** adopts (recognizing that they can't customize their `View()` behaviour for every new class that comes along).

**Comment by RMHogervorst** [comment]
good point @ateucher! Perhaps a general - ignore this column - attribute could be added? One that package creaters could activate .onload() ?

**Comment by jmcphers** [comment]
This issue hasn't been updated in well over a year and lacks a reproducible example. If you're still seeing it on 1.2.5019 (which has many improvements to e.g. managing long column lists), please feel free to reopen with a reproducible example.

**Comment by see24** [comment]
I am still having this issue with Rstudio version 1.2.5033. As mentioned above it only occurs for sf objects with geometry columns that are very complex. 
Here is an example of a file with this issue:
[testView.zip](https://github.com/rstudio/rstudio/files/5625097/testView.zip)

Unzip the file in your working directory and then run: 

``` 
    library(sf)
    testView <- st_read("testView.shp")
    View(testView)
```

**Comment by see24** [comment]
@jmcphers not sure now to reopen but please see the above example

**Comment by kevinushey** [comment]
Here's a stand-alone reproducible example of the same form:

```
library(tibble)

rows <- 20
n <- 1E4

tbl <- tibble(
  a = seq_len(rows),
  b = replicate(rows, as.numeric(seq_len(n)), simplify = FALSE)
)

View(tbl)
```

To fix this, we would want to show a short summary of the list-column's contents rather than the entirety of the deparsed column.
